examples/newrelic/README.md

```diff
@@ -0,0 +1,193 @@
+
+If you are using a Salt based KUBERNETES\_PROVIDER (**gce**, **vagrant**, **aws**), you should enable the
+ability to create privileged containers via the API.
+
+```sh
+$ cd kubernetes
+$ vi cluster/saltbase/pillar/privilege.sls
+
+# If true, allow privileged containers to be created by API
+allow_privileged: true
+```
+
+### Step 1: Configure NewRelic
+
+NewRelic is configured via environment variables. We will configure these environment variables in a sourced bash script, encode the environment file data, and store it in a secret which will be loaded at container runtime.
+
+The [New Relic Linux Server configuration page]
+(https://docs.newrelic.com/docs/servers/new-relic-servers-linux/installation-configuration/configuring-servers-linux) lists all the other settings for nrsysmond.
+
+To create an environment variable for a setting, prepend NRSYSMOND_ to its name. For example,
+
+```sh
+loglevel=debug
+```
+translates to
+
+```sh
+NRSYSMOND_loglevel=debug
+```
+
+Edit examples/newrelic/nrconfig.
+
+<!-- BEGIN MUNGE: EXAMPLE nrconfig -->
+
+```sh
+export NRSYSMOND_loglevel=debug
+export NRSYSMOND_license_key=REPLACE_WITH_REAL_KEY
+```
+
+[Download example](nrconfig?raw=true)
+<!-- END MUNGE: EXAMPLE nrconfig.yaml -->
+
+Be sure to edit the license key field and fill in your own NewRelic license key.
+
+Now, let's vendor the config into a secret.
+
+```sh
+cd examples/newrelic/
+./config-to-secret.sh
+```
+
+<!-- BEGIN MUNGE: EXAMPLE newrelic-secret-template.yaml -->
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: newrelic-keys
+type: Opaque
+data:
+  licensekey: {{license_key_data}}
+```
+
+[Download example](newrelic-secret-template.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE newrelic-secret.yaml -->
+
+The script will encode the config file and write it to `newrelic-config.yaml`.
+
+Finally, submit the config to the cluster:
+
+```sh
+kubectl create -f examples/newrelic/newrelic-config.yaml
+```
+
+### Step 2: Create the DaemonSet definition.
+
+The DaemonSet definition instructs Kubernetes to place a newrelic sysmond daemon on each Kubernetes node.
+
+<!-- BEGIN MUNGE: EXAMPLE newrelic-daemonset.yaml -->
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  labels:
+    name: newrelic
+  name: newrelic
+spec:
+  template:
+    metadata:
+      labels:
+        name: newrelic
+    spec:
+      # Filter to specific nodes:
+      # nodeSelector:
+      #  app: newrelic
+      securityContext:
+        privileged: true
+      hostNetwork: true
+      containers:
+        - resources:
+            limits:
+              cpu: 0.1
+          env:
+            - name: NRSYSMOND_logfile
+              value: "/var/log/nrsysmond.log"
+          image: newrelic/nrsysmond
+          name: newrelic
+          command: [ "bash", "-c", "source /etc/kube-newrelic/config && /usr/sbin/nrsysmond -E -F" ]
```

Comment:
Yep it's valid. For whatever reason I couldn't get it working using the other syntax.